swagger.yaml

```diff
@@ -333,4 +778,235 @@ components:
+          readOnly: true
+        marketValueTotal:
+          type: string
+          pattern: /^[+-]?([0-9]+\.?[0-9]*|\.[0-9]+)$/
+          description: Market total amount in cents
+          readOnly: true
+        purchaseTotal:
+          type: string
+          pattern: /^[+-]?([0-9]+\.?[0-9]*|\.[0-9]+)$/
+          description: Purchase total amount in cents
+          readOnly: true
+
+    PortfolioHistoryValue:
+      type: object
+      description: Portfolio history object
+      properties:
+        date:
+          type: string
+          format: date
+          description: History entry date
+        value:
+          type: integer
+
+    PortfolioSummary:
+      type: object
+      description: Portfolio statistics object
+      required:
+        - subscriptions
+        - redemptions
+      properties:
+        subscriptions:
+          type: string
+          pattern: /^[+-]?([0-9]+\.?[0-9]*|\.[0-9]+)$/
+          description: Subscriptions in cents
+          readOnly: true
+        redemptions:
+          type: string
+          pattern: /^[+-]?([0-9]+\.?[0-9]*|\.[0-9]+)$/
+          description: Redemptions in cents
+          readOnly: true
+
+    PortfolioFund:
+      type: object
+      description: Portfolio fund object
+      required:
+        - id
+        - amount
+        - purchaseValue
+        - totalValue
+      properties:
+        id:
+          type: string
+          format: uuid
+          description: Fund ID. Can be used to search fund details.
+          readOnly: true
+        amount:
+          type: number
+          format: double
+          description: Amount of funds
+          readOnly: true
+        purchaseValue:
+          type: integer
+          format: int64
+          description: Purchase value in cents
+          readOnly: true
+        totalValue:
+          type: integer
+          format: int64
+          description: Total value in cents
+          readOnly: true
+
+    PortfolioTransaction:
+      type: object
+      description: Portfolio transaction object
+      required:
+        - fundId
+        - transactionType
+        - valueDate
+        - value
+        - shareAmount
+        - marketValue
+        - totalValue
+        - paymentDate
+        - provision
+      properties:
+        id:
+          type: string
+          format: uuid
+          description: Portfolio transaction ID in UUID format
+          readOnly: true
+        fundId:
+          type: string
+          format: uuid
+          description: Fund ID
+          readOnly: true
+        targetFundId:
+          type: string
+          format: uuid
+          description: Target fund ID. Only available when transaction type is security
+          readOnly: true
+        transactionType:
+          $ref: "#/components/schemas/TransactionType"
+        valueDate:
+          type: string
+          format: date
+          description: Date of fund value
+          readOnly: true
+        value:
```

Comment:
Should this field be string?

Reply:
maybe, I havent reached to this part of the implementation. I check.